### PR TITLE
Test AGP 8.0 using java 17

### PR DESCRIPTION
**Pull request by reidbaker**
- Update Gradle/AGP version and add namespace plus dependencies. 

https://github.com/flutter/flutter/issues/125181

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by reidbaker** [comment]
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8783181311295663521/+/u/run_test.dart_for_build_tests_shard_and_subshard_1_3/test_stdout
```
Caused by: com.android.builder.errors.EvalIssueException: Android Gradle plugin requires Java 17 to run. You are currently using Java 11."
```

File: packages/integration_test/android/build.gradle
Comment (stuartmorgan) [inline packages/integration_test/android/build.gradle:29]: I know almost nothing about Groovy, so maybe this is an irrelevant difference, but I've been doing using the syntax `namespace 'dev.flutter.integration_test'`, which is what https://developer.android.com/build/configure-app-module#groovy shows for Groovy.

I see looking around more that https://developer.android.com/build/publish-library/prep-lib-release does the `=`, but maybe there's a difference in how unsupported/unknown things are handled with one syntax vs. the other?

Comment (reidbaker) [reply packages/integration_test/android/build.gradle:29]: yeah given that groovy is a dynamic language I wonder if the `=` is more strict.  That seems like a reasonable guess.

File: packages/integration_test/android/build.gradle
Comment (stuartmorgan) [inline packages/integration_test/android/build.gradle:28]: You've dropped the `namespace`; without that people still can't use AGP 8.0 in any project that depends on `integration_test`. That's the one change we absolutely *do* need to make for compatibility, based on my testing.

**Review by stuartmorgan-g — CHANGES_REQUESTED** [review]

File: packages/integration_test/android/build.gradle
Comment (gmackall) [reply packages/integration_test/android/build.gradle]: My understanding of groovy is that `foo 'bar'` is equivalent to `foo('bar')` (i.e. a method call), while `foo = 'bar'` is what you would expect it to be. 

But that they end up equivalent in some places because that `foo` is just a setter method for a `foo` property.

See https://stackoverflow.com/questions/22759542/confused-with-the-use-of-assignment-and-brackets (the question is somewhat different but the top answer covers the topic).

File: packages/integration_test/android/build.gradle
Comment (reidbaker) [reply packages/integration_test/android/build.gradle:28]: fixed

**Review by stuartmorgan-g — APPROVED** [review]
LGTM, with one reservation. I'll leave it to your judgement though.

File: packages/integration_test/example/android/app/build.gradle
Comment (stuartmorgan) [inline packages/integration_test/example/android/app/build.gradle:35]: Apparently I didn't save the comment that was supposed to go with that overall comment:

Could we make *just* this change in one PR, to make it as safe as possible to cherry-pick, and do the rest in a second PR that we don't cherry-pick? I'm slightly worried that the AGP update here might cause unexpected issues for clients (since IIRC we've hit issues in the past where we thought that wouldn't affect clients and then it did, but maybe I'm misremembering).

Comment (reidbaker) [reply packages/integration_test/example/android/app/build.gradle:35]: https://github.com/flutter/flutter/pull/125657

File: packages/flutter_tools/test/integration.shard/android_plugin_compilesdkversion_mismatch_test.dart
Comment (stuartmorgan) [inline packages/flutter_tools/test/integration.shard/android_plugin_compilesdkversion_mismatch_test.dart:26]: It looks like you ran `dart format` on this file, which should be undone. E.g., the breaks after `=` below violate flutter/flutter style (😒).

**Review by stuartmorgan-g — APPROVED** [review]
LGTM modulo the Dart formatting.

File: packages/flutter_tools/test/integration.shard/android_plugin_compilesdkversion_mismatch_test.dart
Comment (reidbaker) [reply packages/flutter_tools/test/integration.shard/android_plugin_compilesdkversion_mismatch_test.dart]: Undone, I keep forgetting we have code that we are not allowed to auto format :/

**Pull request merged**
